src/v/cluster/topic_properties.cc

```diff
@@ -150,6 +153,8 @@ bool topic_properties::requires_remote_erase() const {
            && !read_replica.value_or(false) && remote_delete;
 }
 
+// TODO(oren): need a check somewhere s.t. if iceberg is enabled we populate
```

Comment:
I did something [like this](https://github.com/redpanda-data/redpanda/blob/90c713a9b1cfbd6252a2821357c3673220950bef/src/v/storage/ntp_config.h#L273) in the past

Users of this property get ntp_config from the raft->log instance

Reply:
nice. i think in this case doing nothing is probably fine. I assume we'll still have something like this once your port is done?

https://github.com/redpanda-data/redpanda/blob/90c713a9b1cfbd6252a2821357c3673220950bef/src/v/datalake/datalake_manager.cc#L297-L301

then just wire the topic config in there?